src/services/refactors/moveToFile.ts

```diff
@@ -218,6 +226,9 @@ function getNewStatementsAndRemoveFromOldFile(
         if (targetFile.statements.length > 0) {
             changes.insertNodesAfter(targetFile, targetFile.statements[targetFile.statements.length - 1], body);
         }
+        else {
+            changes.insertNodesAtEndOfFile(targetFile, body, /*blankLineBetween*/ false);
+        }
         if (imports.length > 0) {
             insertImports(changes, targetFile, imports, /*blankLineBetween*/ true, preferences);

```

Comment:
Regarding the extra new lines, we could also change this code to pass `blankLineBetween: false`, and the consequence would be that we'd sometimes not have an empty line between the import blocks and the moved statements in the target file, e.g.:
```ts
import { a } from "./a";
const x = 2;
```
instead of

```ts
import { a } from "./a";

const x = 2;
```